http/src/request/guild/create_guild.rs

```diff
@@ -170,6 +301,32 @@ impl<'a> CreateGuild<'a> {
         self
     }
 
+    /// Set overrides to the everyone role.
+    pub fn override_everyone(mut self, fragment: RoleFragment) -> Self {
+        let everyone = Role {
+            color: fragment.color.unwrap_or(0),
+            hoist: fragment.hoist,
+            id: RoleId(1),
+            managed: false,
+            mentionable: fragment.mentionable,
+            name: fragment.name.unwrap_or_else(|| String::from("@everyone")),
+            permissions: fragment.permissions.unwrap_or_else(Permissions::empty),
+            position: fragment.position.unwrap_or(0),
+        };
+
+        let mut roles: Vec<Role> = vec![everyone];
+
+        // if some roles have already been set, retain them
+        if self.fields.roles.is_some() {
```

Comment:
An `if let` can be used here on `self.fields.roles` to avoid the inner unwrap. Then you can replace the first element or set the roles if it doesn't exist

Reply:
I run into ownership issues here, which is why there is a call to `take`